### PR TITLE
Fix PowerCone in MatrixOfConstraints

### LINKED ISSUE #1721: Bug in PowerCones with MatrixOfConstraints
The ConstraintSet is mangled somewhere:
```Julia
julia> import MathOptInterface

julia> const MOI = MathOptInterface
MathOptInterface

julia> MOI.Utilities.@product_of_sets(Cones, MOI.PowerCone{T})

julia> const OptimizerCache{T} = MOI.Utilities.GenericModel{
           Cdouble,
           MOI.Utilities.ObjectiveContainer{Cdouble},
           MOI.Utilities.VariablesContainer{Cdouble},
           MOI.Utilities.MatrixOfConstraints{
               Cdouble,
               MOI.Utilities.MutableSparseMatrixCSC{
                   Cdouble,
                   T,
                   MOI.Utilities.ZeroBasedIndexing,
               },
               Vector{Cdouble},
               Cones{Cdouble},
           },
       }
MathOptInterface.Utilities.GenericModel{Float64, MathOptInterface.Utilities.ObjectiveContainer{Float64}, MathOptInterface.Utilities.VariablesContainer{Float64}, MathOptInterface.Utilities.MatrixOfConstraints{Float64, MathOptInterface.Utilities.MutableSparseMatrixCSC{Float64, T, MathOptInterface.Utilities.ZeroBasedIndexing}, Vector{Float64}, Cones{Float64}}} where T

julia> model = MOI.Utilities.Model{Float64}()
MOIU.Model{Float64}

julia> x = MOI.add_variables(model, 3)
3-element Vector{MathOptInterface.VariableIndex}:
 MathOptInterface.VariableIndex(1)
 MathOptInterface.VariableIndex(2)
 MathOptInterface.VariableIndex(3)

julia> f = MOI.Utilities.operate(vcat, Float64, 1.0 .* x...)
MathOptInterface.VectorAffineFunction{Float64}(MathOptInterface.VectorAffineTerm{Float64}[MathOptInterface.VectorAffineTerm{Float64}(1, MathOptInterface.ScalarAffineTerm{Float64}(1.0, MathOptInterface.VariableIndex(1))), MathOptInterface.VectorAffineTerm{Float64}(2, MathOptInterface.ScalarAffineTerm{Float64}(1.0, MathOptInterface.VariableIndex(2))), MathOptInterface.VectorAffineTerm{Float64}(3, MathOptInterface.ScalarAffineTerm{Float64}(1.0, MathOptInterface.VariableIndex(3)))], [0.0, 0.0, 0.0])

julia> c = MOI.add_constraint(model, f, MOI.PowerCone(0.2))
MathOptInterface.ConstraintIndex{MathOptInterface.VectorAffineFunction{Float64}, MathOptInterface.PowerCone{Float64}}(1)

julia> cache = OptimizerCache{Int}()
MOIU.GenericModel{Float64, MOIU.ObjectiveContainer{Float64}, MOIU.VariablesContainer{Float64}, MOIU.MatrixOfConstraints{Float64, MOIU.MutableSparseMatrixCSC{Float64, Int64, MOIU.ZeroBasedIndexing}, Vector{Float64}, Cones{Float64}}}

julia> index_map = MOI.copy_to(cache, model)
MathOptInterface.Utilities.IndexMap with 4 entries:
  VariableIndex(1)                                                      => VariableIndex(1)
  VariableIndex(2)                                                      => VariableIndex(2)
  VariableIndex(3)                                                      => VariableIndex(3)
  ConstraintIndex{VectorAffineFunction{Float64}, PowerCone{Float64}}(1) => ConstraintIndex{VectorAffineFunction{Float64}, PowerCone{Float64}}(1)

julia> c_new = index_map[c]
MathOptInterface.ConstraintIndex{MathOptInterface.VectorAffineFunction{Float64}, MathOptInterface.PowerCone{Float64}}(1)

julia> MOI.get(cache, MOI.ConstraintSet(), c_new)
MathOptInterface.PowerCone{Float64}(3.0)
```

This is causing failures in SCS: 
https://github.com/jump-dev/SCS.jl/blob/7150274b37f15490db23318c93c99eef476407d0/test/MOI_wrapper.jl#L73-L77

### PR DESCRIPTION
**Pull request by odow**
Closes #1721 

To be used by https://github.com/jump-dev/SCS.jl/pull/244

### CODE REVIEW

File: src/Utilities/matrix_of_constraints.jl [hunk 1]
```diff
@@ -523,6 +526,19 @@ function load_constants(
     return
 end
 
+load_constants(::Vector, ::Any, ::MOI.AbstractVectorSet) = nothing
```
Comment (matbesancon) [inline src/Utilities/matrix_of_constraints.jl:529]: can be in another PR but shouldn't `load_constants` be documented or underscored?

Comment (odow) [reply src/Utilities/matrix_of_constraints.jl:529]: It already is
https://github.com/jump-dev/MathOptInterface.jl/blob/9ef6efefc4610eb44e4dd7fc1deea2a538b13fa0/src/Utilities/matrix_of_constraints.jl#L175-L192

**Comment by odow** [comment]
SCS is passing tests with this PR, so I think we're good to go.

File: src/Utilities/matrix_of_constraints.jl (hunk 1 above)
Comment (matbesancon) [reply src/Utilities/matrix_of_constraints.jl:529]: my bad, hadn't seen it!

**Review by matbesancon — APPROVED** [review]

**Comment by odow** [comment]
@blegat bump

**Review by blegat — APPROVED** [review]

File: src/Utilities/matrix_of_constraints.jl (hunk 1 above)
Comment (blegat) [inline src/Utilities/matrix_of_constraints.jl:529]: It would probably be safer to be opt-in rather than opt-out here.

Comment (odow) [reply src/Utilities/matrix_of_constraints.jl:529]: That's breaking. And it's a bit of a pain to have to enumerate everything. Really, the failing tests in SCS should have picked this up earlier. But I obviously excluded them during development and didn't revisit them :(

**Pull request merged**